coverage.c

```diff
@@ -0,0 +1,632 @@
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.  */
+
+/* This program calculates coverage from multiple BAMs
+ * simutaneously, to achieve random access and to use the BED interface.
+ * To compile this program separately, you may:
+ *
+ *   gcc -g -O2 -Wall -o bamcov -D_MAIN_BAMCOV coverage.c -lhts -lz
+ */
+
+// C headers
+#include <config.h>
+
+#include <ctype.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdarg.h>  // variadic functions
+#include <limits.h>  // INT_MAX
+#include <math.h>    // round
+#include <stdbool.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+
+#include "htslib/sam.h"
+#include "htslib/hts.h"
+#include "samtools.h"
+#include "sam_opts.h"
+
+const char *VERSION = "0.1";
+
+typedef struct {  // auxiliary data structure to hold a BAM file
+    samFile *fp;     // file handle
+    bam_hdr_t *hdr;  // file header
+    hts_itr_t *iter; // iterator to a region - NULL for us by default
+    int min_mapQ;    // mapQ filter
+    int min_len;     // length filter
+    unsigned int n_reads;  // records the number of reads seen in file
+    unsigned int n_selected_reads; // records the number of reads passing filter
+    unsigned long summed_mapQ; // summed mapQ of all reads passing filter
+    int fail_flags;
+    int required_flags;
+} bam_aux_t;
+
+typedef struct {  // auxiliary data structure to hold stats on coverage
+    unsigned long long n_covered_bases;
+    unsigned long long summed_coverage;
+    unsigned long long summed_baseQ;
+    unsigned long long summed_mapQ;
+    unsigned int n_reads;
+    unsigned int n_selected_reads;
+    int32_t tid;    // chromosome ID, defined by header
+    int beg;
+    int end;
+    int bin_width;
+} stats_aux_t;
+
+#if __STDC_VERSION__ >= 199901L
+#define VERTICAL_LINE "\u2502" // BOX DRAWINGS LIGHT VERTICAL
+
+// UTF8 specifies block characters in eights going from \u2581 (lower one eight block) to \u2588 (full block)
+//   https://en.wikipedia.org/wiki/Block_Elements
+// LOWER ONE EIGHTH BLOCK … FULL BLOCK
+static const char *const BLOCK_CHARS8[8] = {"\u2581", "\u2582", "\u2583", "\u2584", "\u2585", "\u2586", "\u2587", "\u2588"};
+// In some terminals / with some fonts not all UTF8 block characters are supported (e.g. Putty). Use only half and full block for those
+static const char *const BLOCK_CHARS2[2] = {"\u2584", "\u2588"};
+
+#else
+
+// Fall back to explicit UTF-8 encodings of the same characters
+#define VERTICAL_LINE "\xE2\x94\x82"
+
+static const char *const BLOCK_CHARS8*[8] = { 
```

Comment:
Looks like there's a typo here.  The older version of gcc on travis tries to compile this and says:
```
gcc -g -O2 -I. -Ihtslib -I./lz4  -c -o coverage.o coverage.c
coverage.c:93:38: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘*’ token
 static const char *const BLOCK_CHARS8*[8] = { 
```

Reply:
Thanks, will fix that